clients/admin-ui/src/features/system/SystemInformationForm.tsx

```diff
@@ -142,6 +152,41 @@ const SystemInformationForm = ({
     createSystemMutationResult.isLoading ||
     customFields.isLoading;
 
+  const datasetSelectOptions = allEnabledDatasets
+    ? allEnabledDatasets.map((ds) => ({
+        value: ds.fides_key,
+        label: ds.name ? ds.name : ds.fides_key,
+      }))
+    : [];
+
+  const legalBasisForProfilingOptions = [
```

Comment:
All of these dropdown options are being recalculated on each component render. Since they never change I recommend wrapping them in `useMemo` hooks so the values aren't re-instantiated on each render.